### ISSUE TITLE
Thinks block is unused

### ISSUE THREAD

**Issue by tmaier** [rubocop/rubocop#1270]
```
lib/tasks/generate_client.rake:20:19: W: [Corrected] Unused method argument - block. If it's necessary, use _ or _block as an argument name to indicate that it won't be used. You can also write as run(*) if you want the method to accept any arguments but don't care about them.
    def self.run(&block)
                  ^^^^^
```

The code is:

``` ruby
    def self.run(&block)
      client = new
      yield(client) if block_given?
      client.run
    end
```

and rubocop corrects it to:

``` ruby
    def self.run(&_block)
      client = new
      yield(client) if block_given?
      client.run
    end
```

It's used in `yield` and `block_given?`

Why am I adding `(&block)` anyway?
This allows me to see immediately, that this method takes a block.

**Comment by joeyAghion** [comment]
I, for one, would like to see Rubocop accept `&block` in this case. (I recognize that the _variable_ is unused but the argument certainly is.)

The style guide uses it liberally. It's a convention, and what are many of Rubocop's cops if not enforced conventions?

**Comment by mikegee** [comment]
I don't see an unused `&block` arg in https://github.com/bbatsov/ruby-style-guide. Do you mean a different style guide?

**Comment by joeyAghion** [comment]
You're right I suppose, that all instances of `&block` in the style guide (except one "bad" example) are eventually used. I meant that the style guide uses the conventional `&block` name.
